app/controllers/manageBusinesses/manage/ManageIncomeSourceDetailsController.scala

```diff
@@ -186,49 +188,65 @@ class ManageIncomeSourceDetailsController @Inject()(val view: ManageIncomeSource
   }
 
 
-  private def getManageIncomeSourceViewModel(sources: IncomeSourceDetailsModel, incomeSourceId: IncomeSourceId, isAgent: Boolean)
-                                            (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Either[Throwable, ManageIncomeSourceDetailsViewModel]] = {
+  private def getManageIncomeSourceViewModel(
+                                              sources: IncomeSourceDetailsModel,
+                                              incomeSourceId: IncomeSourceId,
+                                              isAgent: Boolean
+                                            )(implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Either[Throwable, ManageIncomeSourceDetailsViewModel]] = {
 
     val desiredIncomeSourceMaybe: Option[BusinessDetailsModel] = sources.businesses
       .filterNot(_.isCeased)
       .find(businessDetailsModel => businessDetailsModel.incomeSourceId == incomeSourceId.value)
 
-    if (desiredIncomeSourceMaybe.isDefined) {
-      itsaStatusService.hasMandatedOrVoluntaryStatusCurrentYear.flatMap {
-        case true =>
-          getCrystallisationInformation(desiredIncomeSourceMaybe.get.latencyDetails).flatMap {
-            case None => Future(Right(variableViewModelSEBusiness(
-              incomeSource = desiredIncomeSourceMaybe.get,
-              itsaStatus = true,
-              crystallisationTaxYear1 = None,
-              crystallisationTaxYear2 = None)))
-            case Some(crystallisationData: List[Boolean]) =>
-              Future(Right(
-                variableViewModelSEBusiness(
-                  incomeSource = desiredIncomeSourceMaybe.get,
-                  itsaStatus = true,
-                  crystallisationTaxYear1 = crystallisationData.headOption,
-                  crystallisationTaxYear2 = crystallisationData.lastOption
-                )))
-          }
-        case false =>
-          Future(Right(
-            variableViewModelSEBusiness(
-              incomeSource = desiredIncomeSourceMaybe.get,
-              itsaStatus = false,
-              crystallisationTaxYear1 = None,
-              crystallisationTaxYear2 = None)
-          ))
-      }
-    } else {
-      Future(Left(
-        new Error("Unable to find income source")
-      ))
+    desiredIncomeSourceMaybe match {
+      case Some(desiredIncomeSource) =>
+        desiredIncomeSource.latencyDetails match {
+          case Some(latencyDetails) =>
+            itsaStatusService.hasMandatedOrVoluntaryStatusForLatencyYears(Some(latencyDetails)).flatMap {
```

Comment:
This has become quite nested, maybe abstract this to it's own function? Or some other way if you think of something better